### PR TITLE
Feature/mrti 3027

### PR DESCRIPTION
**Pull request by Hugo-C**
add_threats now use the API bulk endpoints if possible

### CODE REVIEW

File: datalake_scripts/engines/post_engine.py [hunk 1]
```diff
@@ -169,31 +171,79 @@ def add_threats(self, atom_list: list, atom_type: str, white: bool, threats_scor
...
+            payload['atom_values'] = batch
+            response = self.datalake_requests(self.url, 'post', self._post_headers(), payload)
+            for hashkey in response.get('hashkeys', []):
+                hashkey_created.add(hashkey)
+
+        nb_threats = len(hashkey_created)
+        if nb_threats > 0:
+            ok_sign = '\x1b[0;30;42m' + '  OK  ' + '\x1b[0m'
+            logger.info(f'Created {nb_threats} threats'.ljust(self.terminal_size - 6, ' ') + ok_sign)
+        else:
```
Comment (jakubbanas) [inline datalake_scripts/engines/post_engine.py:241]: Maybe it would be nice to return the errors list in that case, so the user would know which file rows are not correct.

Comment (jakubbanas) [reply datalake_scripts/engines/post_engine.py:241]: Also, when some values in a batch would be not correct it would be difficult to say which atom was created, and which not.

Let's say we have 300 IPs, and one of those is invalid - in that case, only 200 will be created, because one batch will be discarded because of this one incorrect IP. In the output, the user will only see hashes, so there is no way to check which values were rejected.

**Review by jakubbanas — CHANGES_REQUESTED** [review]

File: datalake_scripts/engines/post_engine.py (hunk 1 above)
Comment (Hugo-C) [reply datalake_scripts/engines/post_engine.py:241]: I don't know what we should do then ?  
If the API reject whole batches we should log failed them I suppose but that's a bit wordy :smile:. Since we keep the insertion order from the file, maybe we should only log a range ? like:
> threats from \<value1\> to \<value2\> failed to be created  

Is that ok for you ?

Comment (jakubbanas) [reply datalake_scripts/engines/post_engine.py:245]: I guess logging failed batches would be enough, good idea!

**Review by jakubbanas — APPROVED** [review]

File: datalake_scripts/engines/post_engine.py [hunk 2]
```diff
@@ -169,31 +171,83 @@ def add_threats(self, atom_list: list, atom_type: str, white: bool, threats_scor
...
+        # Build score payload
+        if is_whitelist:
+            for threat in self.authorized_threats_value:
+                payload['scores'].append({'score': {'risk': 0}, 'threat_type': threat})
+        else:
+            for threat, score in threats_score.items():
+                payload['scores'].append({'score': {'risk': score}, 'threat_type': threat})
+
+        for batch in split_list(atom_list, self._batch_size()):
+            payload['atom_values'] = batch
```
Comment (joginsky) [inline datalake_scripts/engines/post_engine.py:232]: This changed in current develop. `atom_values` is not a list anymore. It's a raw content of a .csv file.

**Review by jziolkowski — APPROVED** [review]
